clippy_utils/src/consts.rs

```diff
@@ -1,3 +1,8 @@
+//! A simple const eval API, for use on arbitrary hir expressions.
+//!
+//! This cannot use rustc's const eval, aka miri, as arbitrary hir expressions cannot be lowered to
+//! executable MIR bodies, so we have to do this instead. If hir expressions gain the ability to be
+//! lowered to executable mir bodies, this should be removed.
```

Comment:
Do we have some issue/PR which we can link to track when it happens

Reply:
Not that I know of, I don't think it's even a goal. That's why I used "if" instead of "when".

Reply:
HIR expressions are, practically, never going to gain that ability. We would have to maintain a translation layer in clippy for this.

Reply:
Shall I add a note to say "but this is practically never planned."?

Reply:
I'd say remove the `If HIR expressions...` bit so it's not interpreted as a temporary measure in that case

Reply:
Done.